pyqtgraph/graphicsItems/ROI.py

```diff
@@ -1653,7 +1662,10 @@ def getArrayRegion(self, arr, img=None, axes=(0, 1), **kwds):
         shape = [(n if i in axes else 1) for i,n in enumerate(arr.shape)]
         mask = mask.reshape(shape)
         
-        return arr * mask
+        if returnMappedCoords:
+            return arr * mask, mappedCoords * mask
```

Comment:
Doesn't make sense to me to mask the coordinates as well--each element in the returned array has a corresponding location in the original data, even if that element value was masked.

Reply:
You're right. I pushed a fix. Let me know your thoughts.

Reply:
I disagree, makes more sense to mask out coordinates. I implemented a different fix that uses a flag value to mark which coordinates were masked in the returned array, this also takes into account that coords is a 3 dimensional array:
```
def maskCoordinates(coords, mask, flag=-1):
    """
    Applies the mask to the coordinates, setting any 0 values in mask
    to -1.

    Parameters
    ----------
    coords : numpy.ndarray, coordinates returned from getArrayRegion
    mask : numpy.ndarray, mask to be applied
    flag : int, value to represent 0 values in mask

    Returns
    -------
    m_coords : numpy.ndarray, masked coordinates

    """
    m_coords = coords.copy()
    m_coords[0] *= mask
    m_coords[1] *= mask
    m_coords[0][mask == 0] = flag
    m_coords[1][mask == 0] = flag

    return m_coords
```